phoonnx/model_manager.py

```
@@ -174,6 +190,172 @@ def refresh_voices(self):
+            model_data = voice_list_data[base_id]
+            display_name = self._generate_display_name(base_id, model_data, engine)
+            
+            if engine == Engine.PIPER:
+                # voice_id bevat "piper_" prefix in dit geval
+                lang_code = model_data.get("language") or voice_id.split('-')[0] 
+            else: # Mimic3 of Fallback
+                lang_code = model_data.get("language", "und")
+
+            if engine == Engine.PIPER:
+                base = "https://huggingface.co/rhasspy/piper-voices/resolve/main/"
+                model_url = base + [a for a in model_data["files"] if a.endswith(".onnx")][0]
```

Ruff (0.14.1)

Prefer `next(...)` over single element slice

Replace with `next(...)`

(RUF015)